crates/metadata-store/src/lib.rs

```diff
@@ -393,15 +397,16 @@ impl WriteRequest {
         }
     }
 
-    fn encode_to_vec(&self) -> Result<Vec<u8>, StorageEncodeError> {
-        let mut buffer = BytesMut::new();
-        // todo: Removing support for BufMut requires an extra copy from BytesMut to Vec :-(
-        StorageCodec::encode(self, &mut buffer)?;
-        Ok(buffer.to_vec())
+    fn encode_to_vec(self) -> Result<Vec<u8>, StorageEncodeError> {
+        let request = grpc::WriteRequest::from(self);
+        Ok(request.encode_to_vec())
     }
 
-    fn decode_from_bytes(mut bytes: Bytes) -> Result<Self, StorageDecodeError> {
-        StorageCodec::decode::<WriteRequest, _>(&mut bytes)
+    fn decode_from_bytes(bytes: Bytes) -> Result<Self, StorageDecodeError> {
+        let result = grpc::WriteRequest::decode(bytes).unwrap();
```

Comment:
I totally overlooked the `unwrap()` call. i will create a follow up PR to fix it.